### PR TITLE
Fix deploy-rs, add Auto documentation

### PR DESCRIPTION
**Pull request by Skarlett**
adds package:
  - **flakeDocsMD**
  - **nixosDocsMD**
  - **generateDocsHTML**

adds flake-module:
  - lynx-docs
 
image

```sh
nix build computerclub#generateDocsHTML
firefox result/site/nixos-options
```

### CODE REVIEW

**Comment by JeremiahSecrist** [comment]
I found some errors when building:
```
trace: warning: The check argument to evalModules is deprecated. Please set config._module.check instead.
trace: nixos-options.md
trace: warning: The check argument to evalModules is deprecated. Please set config._module.check instead.
trace: warning: The type `types.string` is deprecated. See https://github.com/NixOS/nixpkgs/pull/66346 for better alternative types.
trace: flake-options.md
```
is this upstream?

**Comment by Skarlett** [comment]
> I found some errors when building:
> 
> ```
> trace: warning: The check argument to evalModules is deprecated. Please set config._module.check instead.
> trace: nixos-options.md
> trace: warning: The check argument to evalModules is deprecated. Please set config._module.check instead.
> trace: warning: The type `types.string` is deprecated. See https://github.com/NixOS/nixpkgs/pull/66346 for better alternative types.
> trace: flake-options.md
> ```
> 
> is this upstream?


# check argument evalModules
```
trace: warning: The check argument to evalModules is deprecated. Please set config._module.check instead.
```
https://github.com/the-computer-club/lynx/blob/64092edcb6a14ebefdee7dd000e243fa0e7b7b6d/flake-modules/lynx-docs/default.nix#L20

Is upstream, but mostly just complaining that I'm not settings the value `_module.check = false` inside each of the modules.

# lingering debug
```
trace: nixos-options.md
```
Is some lingering debug statements, which can be removed.

# Depreciated types
```
trace: warning: The type types.string is deprecated. See https://github.com/NixOS/nixpkgs/pull/66346 for better alternative types.
```

Is derived from `flake-modules/deploy-rs` can be corrected. I'll see if I can quiet it down.

@washedupdeveloper Should be notified that `flake-modules/deploy-rs` only performs type checking now.

**Comment by Skarlett** [comment]
All warnings removed.

File: flake-modules/lynx-docs/default.nix [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+          ''
+            cat ${docs.optionsCommonMark} \
+              ${if htmlSafe then
+                ''| sed -e "s|\\\<|\&lt\;|g"'' else "" } \
+              ${if uri != null then
+                ''| sed -e "s|file:///nix/store/.*-source/|${uri}|g"''
+                else ""} \
+              > $out
+          ''
+      );
```
Comment (Skarlett) [inline flake-modules/lynx-docs/default.nix:40]: I'll rework this hack into flake-module options.

Comment (Skarlett) [reply flake-modules/lynx-docs/default.nix:40]: Also worth noting that breaking apart the htmlSafe & relinking phases into a separate drv would be ideal

**Review by JeremiahSecrist — APPROVED** [review]

**Pull request merged**